### PR TITLE
mask rcnn guide

### PR DESCRIPTION
**Pull request by priyankatuteja**
Mask RCNN concept guide

### CODE REVIEW

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb [hunk 1]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (Yongyao) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <ol><li>For those bold text, shall we keep the lowercase/uppercase consistent? Right now, it is "&nbsp;<strong>Object Detection" and "Object instance segmentation".</strong></li><li><strong>Can we add hyperlink for "how unet works" as well?</strong></li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Yongyao) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>Perhaps we can add a ground truth image here as well? Right now, it's hard to tell whether these are different buildings or the same. Or perhaps just adding some explanation is enough</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (Yongyao) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>The resolution of this screenshot doesn't look very good.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (Yongyao) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>Are you saying "In the second stage of Mask R-CNN"? It's a bit confusing here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (Yongyao) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>Are you planning on adding an implementation in python api section here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (priyankatuteja) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>I discussed this with Rohit. He suggested that since MaskRCNN does not have any parameters like backbone, pretrained that needs explaination..so we may even remove this section.</p>

Comment (priyankatuteja) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>No, I am talking about Faster RCNN only. In my understanding, MaskRCNN is made of Faster R-CNN + Mask Head where Faster R-CNN's second stage has a RoIAlign layer instaed of RoIPool.</p>

Comment (priyankatuteja) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>The initial size of the image was big. And it looked odd to me,, So I have decreased the size which has caused this image to look blurry. Please have a look at the same image in <a href="https://github.com/Esri/arcgis-python-api/blob/3adbc83829584f9b6aa0197cd8272240ab0ff0a1/guide/14-deep-learning/How_Instance_segmentation_MaskRCNN_works.ipynb" target="_blank">https://github.com/Esri/arcgis-python-api/blob/3adbc83829584f9b6aa0197cd8272240ab0ff0a1/guide/14-deep-learning/How_Instance_segmentation_MaskRCNN_works.ipynb</a>. If this size looks fine to you, I can replace it back.</p>

Comment (priyankatuteja) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>The link yoou had shared i.e. <a href="https://www.analyticsvidhya.com/blog/2019/07/computer-vision-implementing-mask-r-cnn-image-segmentation/" target="_blank">https://www.analyticsvidhya.com/blog/2019/07/computer-vision-implementing-mask-r-cnn-image-segmentation/</a> also shows comparison without showing ground truth, It looks fine to me. I am editing labels a it which may give more clarity.</p>

**Comment by priyankatuteja** [comment]
@Yongyao Thanks for the quick response and review.

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (AtmaMani) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>"precisely segmenting <strong>a </strong>mask for each object instance." (added article 'a')</p><p> </p><p>Another use case is in case of Self Driving Cars -&gt; user lower case</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (AtmaMani) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>This graphic is very helpful</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (AtmaMani) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>Mask R-CNN is a state of the art model for instance segmentation, developed on top of Faster R-CNN. Faster R-CNN is <strong>a</strong> region-based convolutional neural network [2], that returns bounding boxes <strong>for</strong> each object and its class label with <strong>a</strong> confidence score</p><p> </p><p>Stage1: ... to give <strong>a </strong>set of region proposals....</p><p>Region proposals <strong>are</strong> regions in <strong>the </strong>feature map which contain <strong>the </strong>object.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/564/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by AtmaMani** [comment]
@priyankatuteja I have requested some changes

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (priyankatuteja) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb]: <p>Thanks!</p>

**Comment by AtmaMani** [comment]
@priyankatuteja thanks for the guide, @Yongyao thanks for the reviews
